src/main/kotlin/com/wgslfuzz/core/Resolver.kt

```diff
@@ -765,15 +821,20 @@ private fun resolveUnary(
         if (pointerType !is Type.Pointer) {
             throw RuntimeException("Dereference applied to expression $expression with non-pointer type")
         }
-        pointerType.pointeeType
+        Type.Reference(pointerType.pointeeType, pointerType.addressSpace, pointerType.accessMode)
     }
 
     UnaryOperator.ADDRESS_OF -> {
-        resolveTypeOfAddressOfExpression(expression, resolverState)
+        val referenceType = resolverState.resolvedEnvironment.typeOf(expression.target)
```

Comment:
Again, I wonder if you might cross-reference the WGSL spec in a comment here. The relationship between pointers and references is subtle.